### PR TITLE
nsc-events-nestjs_Sprint-5_Restrict-Access-to-User-Routes_34

### LINKED ISSUE #34: Restrict Access to User Routes
An admin role should be the only role allowed to access any user routes.

See Tutorials in the wiki.
UseGuard() and AuthGuard() should be the implemented decorators. You can also view the implementation in the other routes in the Event controller.

### PR DESCRIPTION
**Pull request by Exochos**
Since all routes can be guarded from a top level authGuard I added the imports and then the authguard on line 20, since we also already have a jwt.strategy written I just used that for the auth. All of this trickles down to all of the below user routes. Proceeded to deleted un-needed authguards. 
EDIT, added tests for these routes.
Closes #34 

Here are tests,
1) use the signup route and use the token retrieved to login to the next routes.
1_signup

2) Attempt to create a user without auth.
2_createuser_without_auth

3) Create a user with auth.
3_create_user_with_auth

4) Attempt to update a user without auth.
4__2_update_user_withoutAuth

5) Attempt to update a user with auth token.
4_update_user

6) Attempt to delete a user without Auth.
5__1_delete_user_withoutAuth

7) Attempt to delete a user with auth.
5__2_delete_user_withAuth

### CODE REVIEW

File: src/user/controllers/user/user.controller.ts [hunk 1]
```diff
@@ -8,49 +8,46 @@ import {
...
 } from '@nestjs/common';
 import { UserService } from '../../services/user/user.service';
 import { Role } from '../../schemas/user.model';
 import { CreateUserDto } from 'src/user/dto/create-user.dto';
+import { AuthGuard } from '@nestjs/passport';
 
 // ================== User admin routes ======================== \\
-// TODO: Add AuthGuard to all admin routes and check for admin roles
 @Controller('users')
+@UseGuards(AuthGuard('jwt'))
```
Comment (theGaryLarson) [inline src/user/controllers/user/user.controller.ts:20]: @Exochos  If we are able to just apply @UseGuards to the entire controller please remove  the commented lines where it is implemented for each endpoint.

**Review by theGaryLarson — DISMISSED** [review]
Hi Jeremy, the code looks good but CI/CD is failing because the test is not implemented. Please uncomment and update the test. Review the activity controller test if needed.

**Comment by Exochos** [comment]
> Hi Jeremy, the code looks good but CI/CD is failing because the test is not implemented. Please uncomment and update the test. Review the activity controller test if needed.

I have uploaded a Jest mock test which should hopefully fix this issue.

**Comment by taylorpapke** [comment]
@theGaryLarson looks like there was an update to this PR. Are the changes requested now satisfied?

**Review by taylorpapke — APPROVED** [review]
Awesome! I ran through all of your screenshots and was able to get the same results. Pretty cool! So it seems like this pretty much gives admins control over user creation, updates and deletes while guarding these actions from other users.

**Review by Robel-003 — APPROVED** [review]
This looks great, and thank you for providing some pictures! 
I'm able to create/update/delete an account using the proper endpoints. 
This is minor and can be added later but I'd recommend providing a message for the update and delete endpoints (primarily for the admin/creator)

**Comment by Exochos** [comment]
> This looks great, and thank you for providing some pictures! I'm able to create/update/delete an account using the proper endpoints. This is minor and can be added later but I'd recommend providing a message for the update and delete endpoints (primarily for the admin/creator)

Yeah I didn't know what best practice was in this case to I just left it with the response 200 code.

**Comment by brinkbrink** [comment]
Looks like some checks still aren't going through with the CI/CD. I just re-ran the tests in case something got hung up (it happens sometimes) but that wasn't the case. Maybe troubleshooting this with @theGaryLarson tonight would be a good idea.

**Comment by Seiyaroo** [comment]
Reopened and fixed the checks. Just need to confirm that @theGaryLarson is satisfied per the pending change request.

**Pull request merged**